app/javascript/shared/components/focusTrap.jsx

```diff
@@ -0,0 +1,30 @@
+import PropTypes from 'prop-types';
+import { h, Fragment } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+import { createFocusTrap } from 'focus-trap';
+import { defaultChildrenPropTypes } from '../../common-prop-types';
+
+const FocusTrap = ({ selector, children }) => {
+  const focusTrap = useRef(null);
+
+  useEffect(() => {
+    focusTrap.current = createFocusTrap(selector);
+    focusTrap.current.activate();
+    return () => {
+      focusTrap.current.deactivate();
+    };
+  });
+
+  return <Fragment>{children}</Fragment>;
+};
+
+FocusTrap.defaultProps = {
+  selector: '.crayons-modal',
```

Comment:
I made the default selector value '.crayon-modal' since this component will be mostly be used by modals (rarely in other places where different selector string could be passed)

Reply:
If you open a second modal over the top of an existing one I think the tabbing will be locked to the first.
That being said I believe, with how you've set it up, you can some something like

```jsx
<FocusTrap selector={reference.current}>
  <div id="locked-in" ref={reference}></div>
</FocusTrap>
```

Reply:
Ref would work too. Incase modal appears another modal it can pass another selector string like id of the modal and it would work.

```
<FocusTrap selector="#secondModal">
  <div>
    second modal
  </div>
</FocusTrap>
```
But on the other hand nested modals might not be used anywhere (but I could be wrong). Having no prop needed to be passed for this component by default makes API much simpler.